sdk/src/main/kotlin/com/processout/sdk/api/network/interceptor/UserAgentInterceptor.kt

```diff
@@ -22,8 +22,13 @@ internal class UserAgentInterceptor(
             "ProcessOut Android-Bindings",
             sdkVersion
         )
-        val userAgentRequest = chain.request().newBuilder()
-            .header("Idempotency-Key", UUID.randomUUID().toString())
+        val request = chain.request()
+        val updatedRequest = request.newBuilder()
+            .apply {
+                if (request.method == "POST") {
+                    header("Idempotency-Key", UUID.randomUUID().toString())
+                }
```

Comment:
`Idempotency-Key` is part of retry mechanism so maybe should be moved to `RetryInterceptor`?

Reply:
Good point, I've pushed an update